public/app/features/dashboard/components/DashExportModal/DashboardExporter.ts

```diff
@@ -116,7 +116,7 @@ export class DashboardExporter {
               pluginName: ds.meta?.name,
             };
 
-            if (obj.datasource === null || typeof obj.datasource === 'string') {
+            if (!obj.datasource || typeof obj.datasource === 'string') {
               obj.datasource = '${' + refName + '}';
             } else {
               obj.datasource.uid = '${' + refName + '}';
```

Comment:
Is there a specific reason why we don't _aways_ do `obj.datasource.uid = '${' + refName + '}';`?

It seems (not knowing much about the rest of things) that upon  importing this dashboard it would be migrated to the `datasource.uid` format anyway.

Reply:
i was also a little unsure, but the previous PR from @torkelo suggests that annotation queries are what's blocking us from just using it for everything: https://github.com/grafana/grafana/pull/42305/files#r756956781